scripts/c_header_guards.py

```diff
@@ -0,0 +1,157 @@
+# - key: path of header files to check;
+# - value: path that must be trimmed from the key value in order to get the
+#          final guard.
+FOLDERS_TO_CHECK = {
+    'drivers/builtin/include/mbedtls/': 'drivers/builtin/include/',
+    'drivers/builtin/include/mbedtls/private/': 'drivers/builtin/include/',
+    'drivers/builtin/src/': 'drivers/builtin/src/',
+    'drivers/everest/include/tf-psa-crypto/private/everest/':
+        'drivers/everest/include/tf-psa-crypto/private/everest/',
+    'drivers/p256-m/': 'drivers/p256-m/',
+    'drivers/p256-m/p256-m/': 'drivers/p256-m/',
+    'include/mbedtls/': 'include/',
+    'include/psa/': 'include/',
+    'include/tf-psa-crypto/': 'include/',
+}
+
+def generate_guard_name(file_path: str, trim_path: str) -> str:
+    """Generate the new header guard."""
+    adjusted_name = file_path
+    # Trim prefix
+    if adjusted_name.startswith(trim_path):
+        adjusted_name = adjusted_name[len(trim_path):]
+    # Remove suffix
+    if adjusted_name.endswith('.h'):
+        adjusted_name = adjusted_name[:-2]
+
+    adjusted_name = re.sub(r'[^a-zA-Z0-9]', '_', adjusted_name).upper()
+    guard_elements = []
+    # Add prefix only to non public header files
+    if file_path.startswith('drivers'):
+        guard_elements.append('TF_PSA_CRYPTO')
+    guard_elements.extend([adjusted_name, 'H'])
+    return '_'.join(guard_elements)
+
+def find_line(lines: List[str], search_range, regex: str) -> Tuple[int, Match]:
+    """Try to match the regex expression on the range of lines given in input.
+    The first successful match is returned; an execption is returned if no
+    match is found in all the given lines.
+
+    Input params:
+        lines: content of the file given as list of strings;
+        search_range: range() to use when going through the list of lines;
+        regex: the regex expression to be used for the match.
+
+    Returns: a tuple where
+        int: the first line where the match was successful;
+        Match: the Match object returned on the matching line.
+    """
+    for i in search_range:
+        stripped = lines[i].strip()
+        match = re.match(regex, stripped)
+        if match is not None:
+            return i, match
+
+    raise ValueError(f'Error: {regex} line not found')
+
+def process_header(file_path, trim_path, fix) -> None:
+    """Check/fix header guard in a single file.
+    On failure an exception is thrown.
+    """
+    with open(file_path, 'r', encoding='utf-8') as f:
+        content = f.read()
+
+    correct_guard = generate_guard_name(file_path, trim_path)
+
+    lines = content.split('\n')
+
+    ifndef_line, match = find_line(lines, range(0, len(lines)),
+                                   r'#ifndef\s+(\w+)\b')
+    expected_line = f'#ifndef {correct_guard}'
+    if expected_line not in lines[ifndef_line]:
+        if fix:
+            lines[ifndef_line] = expected_line
+        else:
+            raise ValueError(f'Invalid guard format:\n'
```

Comment:
Note: I'm not a big fan of using exceptions for normal errors, but I guess it's a matter of taste/style, and considering this script is not central, I'm not going to argue about this.